### PR TITLE
make textarea counter work when value is prefilled

### PR DESCRIPTION
**Pull request by caravinci**
**Description of changes**

these changes ensure the textarea counter is aware of values being pre-filled upon component mount.

**Type of Change**

- [x] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [x] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [ ] Feature is minimally responsive
- [x] Manual testing of UI feature
- [ ] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: uikit/form/Textarea/index.tsx [hunk 1]
```diff
@@ -116,12 +113,22 @@ const Textarea = ({
...
+    // normalise line breaks
+    let targetValue = event.target.value.replace(/(\r\n|\r|\n)/g, '\n');
+
+    applyChanges(targetValue);
+    propsOnChange?.(event);
+  };
+
+  useEffect(() => {
+    value && applyChanges(value);
+  }, []);
```
Comment (caravinci) [inline uikit/form/Textarea/index.tsx:130]: apply the counter logic upon mount, if there's a preloaded value.

File: uikit/form/Textarea/index.tsx [hunk 2]
```diff
@@ -116,12 +113,22 @@ const Textarea = ({
...
     [countLimit, propsOnChange, truncate],
   );
 
+  const handleChange = (event) => {
+    // normalise line breaks
+    let targetValue = event.target.value.replace(/(\r\n|\r|\n)/g, '\n');
+
+    applyChanges(targetValue);
+    propsOnChange?.(event);
+  };
```
Comment (caravinci) [inline uikit/form/Textarea/index.tsx:126]: and ensure its still applied if the textarea changes

File: uikit/form/Textarea/index.tsx [hunk 3]
```diff
@@ -65,16 +65,13 @@ const Textarea = ({
...
-  const onChange = useCallback(
-    (event) => {
-      // normalise line breaks
-      let targetValue = event.target.value.replace(/(\r\n|\r|\n)/g, '\n');
+  const getCount = useCallback((newCount) => (isAscending ? newCount : countLimit - newCount), [
+    countLimit,
+    isAscending,
+  ]);
 
+  const applyChanges = useCallback(
```
Comment (caravinci) [inline uikit/form/Textarea/index.tsx:73]: just abstract the change logic to be used not only on textarea change (which turns out doesn't get triggered when the value is preloaded by React on component mount)

File: uikit/form/Textarea/index.tsx [hunk 4]
```diff
@@ -65,16 +65,13 @@ const Textarea = ({
...
-  );
-
-  const onChange = useCallback(
-    (event) => {
-      // normalise line breaks
-      let targetValue = event.target.value.replace(/(\r\n|\r|\n)/g, '\n');
+  const getCount = useCallback((newCount) => (isAscending ? newCount : countLimit - newCount), [
+    countLimit,
+    isAscending,
+  ]);
```
Comment (caravinci) [inline uikit/form/Textarea/index.tsx:71]: prettier being a stickler

**Review by caravinci — COMMENTED** [review]
bit of a walkthrough

**Comment by caravinci** [comment]
This PR belonged to https://github.com/icgc-argo/dac-ui/issues/163
Reconnecting to its new own ticket https://github.com/icgc-argo/dac-ui/issues/313.

File: uikit/form/Textarea/index.tsx [hunk 5]
```diff
@@ -116,12 +113,22 @@ const Textarea = ({
...
     },
     [countLimit, propsOnChange, truncate],
   );
 
+  const handleChange = (event) => {
+    // normalise line breaks
+    let targetValue = event.target.value.replace(/(\r\n|\r|\n)/g, '\n');
+
+    applyChanges(targetValue);
+    propsOnChange?.(event);
```
Comment (mistryrn) [inline uikit/form/Textarea/index.tsx:125]: Never seen this before, but I really like it! Will definitely be using this going forward :)

**Review by mistryrn — APPROVED** [review]
Looks great! 👍

File: uikit/form/Textarea/index.tsx (hunk 5 above)
Comment (caravinci) [reply uikit/form/Textarea/index.tsx:125]: Nifty, right? here's some reference. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Pull request merged**